packages/material-ui-system/src/index.d.ts

```diff
@@ -131,3 +131,9 @@ export { SpacingOptions, Spacing } from './createTheme/createSpacing';
 
 export { default as shape } from './createTheme/shape';
 export * from './createTheme/shape';
+
+export { default as unstable_useThemeProps } from './useThemeProps';
+export { getThemeProps as unstable_getThemeProps } from './useThemeProps';
```

Comment:
Is it still unstable?

Reply:
I would open in the end PR for stabilizing these utils :) we can then decide what we want to be publicly available. I would say `useThemeProps` should be, for `getThemeProps` I am not sure :)

Reply:
I remember advocating something about it in the past but I can't remember what nor where. Currently, if feels that we only need one helper not two. I wonder if:

```jsx
const { props, theme, isRtl } = useThemeProps({ props: inProps, name: 'MuiAccordion' });
```

wouldn't do the job.

In most cases, it seems that spreading `isRTL` and `theme` to the props is not necessary.

Reply:
Regarding the `getThemeProps` I left it, as it existed even before (and it's used in the `useMediaQuery` util if I am not mistaken.

Reply:
> It existed even before

@mnajdova I think that `getThemeProps` was created before the hook existed. We couldn't bundle the theme inside it. 

> How about we remove the theme and isRtl?

Sounds good, It was the direction taken in https://github.com/mui-org/material-ui/pull/26649#discussion_r649569406 by moving the theme and isRtl out of the props. 

My question would then be. How would you merge `useThemeProps` and `getThemeProps` into a single helper?

Reply:
I see there is a follow-up in #26701. No need to continue the discussion here.